security/login_link.rst

```diff
@@ -0,0 +1,510 @@
+information changes (e.g. their e-mailadres).
+
+The signed URL contains 3 parameters:
+
+``expires``
+    The UNIX timestamp when the link expires.
+
+``user``
+    The value returned from ``$user->getUsername()`` for this user.
+
+``hash``
+    A hash of ``expires``, ``user`` and any configured signature
+    properties. Whenever these change, the hash changes and previous login
+    links are invalidated.
+
+You can add more properties to the ``hash`` by using the
+``signature_properties`` option:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            firewalls:
+                main:
+                    login_link:
+                        check_route: login_check
+                        signature_properties: ['email']
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <config>
+                <firewall name="main">
+                    <login-link check-route="login_check">
+                        <signature-property>email</signature-property>
+                    </login-link>
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                'main' => [
+                    'login_link' => [
+                        'check_route' => 'login_check',
+                        'signature_properties' => ['email'],
+                    ],
+                ],
+            ],
+        ]);
+
+The properties are fetched from the user object using the
+:doc:`PropertyAccess component </components/property_access>` (e.g. using
+``getEmail()`` or a public ``$email`` property in this example).
+
+.. tip::
+
+    You can also use the signature properties to add very advanced
+    invalidating logic to your login links. For instance, if you store a
+    ``$lastLinkRequestedAt`` property on your users that you update in the
+    ``requestLoginLink()`` controller, you can invalidate all login links
+    whenever a user requests a new link.
+
+Only allow a Link to be used Once
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is a common characteristic of login links to invalidate as soon as they
+are used. Symfony can support this by storing used login links in the
+cache. Enable this support by setting the ``max_uses`` option:
```

Comment:
Site question: what happened if the cache is cleared? Links are available again?

Reply:
Yea, this would be an important thing to note: you need to make sure your cache doesn't run out of space, else invalid links will become valid. However, the lifetime of the cache items is quite short.

About clearing the cache, yes, the links would become available again. But you would need to manually do a `cache:pool:clear name_of_cache_pool` to do that - it wouldn't happen on a normal `cache:clear`.